File: Py/8492_cm.py
import sys

n = int(sys.stdin.readline().strip())

# n * 2인 복도를 채우는 방법

# n이 1 이상이면 + 1이므로 기본으로 1
# 만약 2로 나눠지면 + 1
# 3이면 3개
# 4면 5개
# 5면 8개
# n = (n-1) + (n-2)

arr = [0] * (n + 1)
arr[1] = 1
arr[2] = 2

if (n <= 2):
	print(arr[n])
	sys.exit()

for i in range(3, n + 1):
	arr[i] = arr[i - 1] + arr[i - 2]

print(arr[n] % 796796)

File: JS/8492_cm.js
const { exit } = require("process");

const n = 5;

arr = new Array(n).fill(0);
arr[1] = 1
arr[2] = 2

if (n <= 2) {
	console.log(arr[n]);
	exit();
}

for (let i = 3; i < n + 1; i++) {
	arr[i] = arr[i - 1] + arr[i - 2]
}

console.log(arr[n] % 796796)
